### PR TITLE
[KHRGA-86] Updated EP h_item classes

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA-86

### CODE REVIEW

File: source/iface/h_item.rst [hunk 1]
```diff
@@ -12,99 +10,218 @@
 
 ::
 
-   template <int dimensions>
-   class h_item;
+  template <int Dimensions>
+  class h_item;
+
+``h_item<int Dimensions>`` identifies an instance of a
```
Comment (mkinsner) [inline source/iface/h_item.rst:16]:
```suggestion
The ``h_item`` class identifies an instance of a
```

File: source/iface/h_item.rst [hunk 2]
```diff
@@ -12,99 +10,218 @@
...
+``group::parallel_for_work_item`` function object executing at each
+point in a local ``range<int Dimensions>`` passed to a
```
Comment (mkinsner) [inline source/iface/h_item.rst:18]:
```suggestion
point in a local ``range`` passed to a
```

File: source/iface/h_item.rst [hunk 3]
```diff
@@ -12,99 +10,218 @@
...
+``parallel_for_work_item`` call or to the corresponding
+``parallel_for_work_group`` call if no ``range`` is passed to the
+``parallel_for_work_item`` call. It encapsulates enough information
+to identify the work-item's local and global items according to the
+information given to ``parallel_for_work_group`` (physical ids) as well
+as the work-item's logical local items in the logical local range.
+All returned items objects are offset-less. Instances of the
```
Comment (mkinsner) [inline source/iface/h_item.rst:25]:
```suggestion
All returned item objects are offset-less. Instances of the
```

File: source/iface/h_item.rst [hunk 4]
```diff
@@ -12,99 +10,218 @@
...
+``h_item<int Dimensions>`` class are not user-constructible and
```
Comment (mkinsner) [inline source/iface/h_item.rst:26]:
```suggestion
``h_item`` class are not user-constructible and
```

File: source/iface/h_item.rst [hunk 5]
```diff
@@ -12,99 +10,218 @@
...
 
 ``get_logical_local``
 =====================
 
 ::
 
-  sycl::item<dimensions, false> get_logical_local() const;
+  item<Dimensions, false> get_logical_local() const
+
+Return the constituent element of the logical local item work-item’s
```
Comment (mkinsner) [inline source/iface/h_item.rst:78]:
```suggestion
Return the constituent element of the logical local work-item’s
```

File: source/iface/h_item.rst [hunk 6]
```diff
@@ -12,99 +10,218 @@
...
+position in the local iteration space as provided upon the invocation
+of the ``group::parallel_for_work_item``.
+
+If the ``group::parallel_for_work_item`` was called without any logical
```
Comment (mkinsner) [inline source/iface/h_item.rst:82]:
```suggestion
If ``group::parallel_for_work_item`` was called without any logical
```

File: source/iface/h_item.rst [hunk 7]
```diff
@@ -12,99 +10,218 @@
...
 
 ``get_physical_local``
 ======================
 
 ::
 
-  sycl::item<dimensions, false> get_physical_local() const;
+  item<Dimensions, false> get_physical_local() const
+
+Return the constituent element of the physical local item work-item’s
```
Comment (mkinsner) [inline source/iface/h_item.rst:97]:
```suggestion
Return the constituent element of the physical local work-item’s
```

File: source/iface/h_item.rst (hunk 2 above)
Comment (imvucic) [reply source/iface/h_item.rst]: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

File: source/iface/h_item.rst (hunk 3 above)
Comment (imvucic) [reply source/iface/h_item.rst]: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

File: source/iface/h_item.rst (hunk 5 above)
Comment (imvucic) [reply source/iface/h_item.rst]: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

File: source/iface/h_item.rst (hunk 6 above)
Comment (imvucic) [reply source/iface/h_item.rst]: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

File: source/iface/h_item.rst (hunk 7 above)
Comment (imvucic) [reply source/iface/h_item.rst]: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

**Pull request merged**